src/meta/meta_bulk_load_service.h

```diff
@@ -404,6 +406,8 @@ class bulk_load_service
     std::unordered_map<app_id, bool> _apps_cleaning_up;
     // Used for bulk load rolling back to downloading
     std::unordered_map<app_id, bool> _apps_rolling_back;
+    // Used for restrict bulk load rollback count
+    std::unordered_map<app_id, int32_t> _apps_rollback_count;
```

Comment:
seem to only need `_apps_rollback_count`,  `_apps_rolling_back` can be replaced by it

Reply:
They have different usage, `_apps_rollback_count` records the rollback count to avoid rollback too many times, `_apps_rolling_back` records if the app is executing rollback to avoid rollback at the same time.

Reply:
you can also judge if the app is rolling back by   `_apps_rollback_count[app_id]` result

Reply:
`_apps_rollback_count` can't replace `_apps_rolling_back`. Let's consume the situtation, replica server A crash during bulk load, partition0 and partition1 on it. If meta server got partition0 timeout first, meta sever will set the app rollback to downloading, then partition1 will raise it again, always happened when this app is executing rolling back, `_apps_rolling_back` is used to solve this condition. 
Well, `_apps_rollback_count` can't do that. If the current count = 2, count=3 indicating app is rolling back, meta server can't judge the rolling back action through this value.